### PR TITLE
Temporarily fix SuperPMI crossgen2 collections

### PR DESCRIPTION
**Pull request by BruceForstall**
Stop calling unimplemented functions when prejitting.

This unfortunately also affects crossgen1.

Tracking issue: https://github.com/dotnet/runtime/issues/48888

### CODE REVIEW

File: src/coreclr/jit/compiler.cpp [hunk 1]
```diff
@@ -3797,28 +3797,28 @@ void Compiler::compSetOptimizationLevel()
 
 #if 0
     // The code in this #if can be used to debug optimization issues according to method hash.
-	// To use, uncomment, rebuild and set environment variables minoptshashlo and minoptshashhi.
+    // To use, uncomment, rebuild and set environment variables minoptshashlo and minoptshashhi.
 #ifdef DEBUG
     unsigned methHash = info.compMethodHash();
     char* lostr = getenv("minoptshashlo");
     unsigned methHashLo = 0;
-	if (lostr != nullptr)
```
Comment (BruceForstall) [inline src/coreclr/jit/compiler.cpp:3805]: This is an unrelated tabs->spaces conversion.

**Comment by BruceForstall** [comment]
@trylek @kunalspathak @dotnet/jit-contrib PTAL

**Review by trylek — APPROVED** [review]
Looks good, thanks for fixing!

**Review by kunalspathak — APPROVED** [review]
